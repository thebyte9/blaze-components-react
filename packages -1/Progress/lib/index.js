"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
exports.__esModule = true;
var react_1 = __importStar(require("react"));
var Progress = function (_a) {
    var progress = _a.progress, steps = _a.steps, type = _a.type, onChange = _a.onChange;
    var _b = react_1.useState(progress), currentProgress = _b[0], setProgress = _b[1];
    var handleClick = function (_a) {
        var event = _a.event, step = _a.step;
        setProgress(step);
        onChange({ event: event, step: step });
    };
    var checkStep = function (step) {
        if (step === steps.length && step === currentProgress) {
            return "final current";
        }
        if (step === currentProgress) {
            return "current";
        }
        if (step === steps.length) {
            return "final";
        }
        if (step <= currentProgress) {
            return "visited";
        }
        return "";
    };
    var isTypeCount = type === "count" ? "progress-bar__list-item--dots" : "";
    var modifiers = function () {
        var allModifiers = type.split(" ");
        var blockElement = "progress-bar__list-item--";
        if (!allModifiers.length) {
            return "" + blockElement + type;
        }
        return allModifiers
            .map(function (modifier) { return "" + blockElement + modifier; })
            .join(" ");
    };
    return (react_1["default"].createElement("nav", { className: "progress-bar" },
        react_1["default"].createElement("ol", { className: "progress-bar__list" }, steps.map(function (text, index) { return (react_1["default"].createElement("li", { key: text, className: "progress-bar__list-item " + modifiers() + " " + isTypeCount + " " + checkStep(index + 1) },
            react_1["default"].createElement("span", { onClick: function (event) { return handleClick({ event: event, step: index + 1 }); }, role: "button" }, text))); }))));
};
Progress.defaultProps = {
    onChange: function () {
        return;
    },
    progress: 0,
    steps: [],
    type: "dots"
};
exports["default"] = Progress;
