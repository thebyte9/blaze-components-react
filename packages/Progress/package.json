{
  "name": "@blaze-react/progress",
  "version": "0.8.0-alpha.66",
  "description": "",
  "main": "./dist/main.js",
  "scripts": {
    "build": "rm -Rf ./dist && NODE_ENV=production webpack --config webpack.config.js",
    "build-storybook": "build-storybook -c .storybook -o docs-build",
    "lint": "eslint '*/**/*.{ts,tsx}' --ignore-path ../../.eslintignore --fix",
    "test": "jest --passWithNoTests",
    "test:coverage": "jest --coverage --passWithNoTests",
    "prepublishOnly": "yarn build",
    "bundle-analyser": "node ../../node_modules/webpack-bundle-analyzer/lib/bin/analyzer.js ./stats/stats.json"
  },
  "keywords": [],
  "author": "byte9",
  "license": "GPL-3.0",
  "devDependencies": {
    "@blaze-react/blaze-components-theme": "^0.8.0-alpha.59"
  },
  "peerDependencies": {
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "webpack": "^5.28.0",
    "webpack-cli": "^4.6.0"
  },
  "gitHead": "d9cac6fc28dad1d0794718aac09d292019446698"
}
