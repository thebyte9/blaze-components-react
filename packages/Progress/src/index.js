import React, { useState } from 'react';
import PropTypes from 'prop-types';

const Progress = ({ progress, steps, type, onChange }) => {
  const [_progress, setProgress] = useState(progress);

  const handleClick = ({ event, step }) => {
    setProgress(step);
    onChange({ event, step });
  };

  const checkStep = step => {
    if (step === steps.length && step === _progress) return 'final current';
    if (step === _progress) return 'current';
    if (step === steps.length) return 'final';
    if (step <= _progress) return 'visited';
    return '';
  };

  const isTypeCount = type === 'count' ? 'progress-bar__list-item--dots' : '';

  const modifiers = () => {
    const _modifiers = type.split(' ');
    const blockElement = 'progress-bar__list-item--';
    if (!_modifiers.length) return `${blockElement}${type}`;
    return _modifiers.map(modifier => `${blockElement}${modifier}`).join(' ');
  };

  return (
    <nav className="progress-bar">
      <ol className="progress-bar__list">
        {steps.map((text, index) => (
          <li
            key={text}
            className={`progress-bar__list-item ${modifiers()} ${isTypeCount} ${checkStep(
              index + 1
            )}`}>
            <span onClick={event => handleClick({ event, step: index + 1 })} role="button">
              {text}
            </span>
          </li>
        ))}
      </ol>
    </nav>
  );
};

Progress.propTypes = {
  progress: PropTypes.number,
  type: PropTypes.string,
  steps: PropTypes.array,
  onChange: PropTypes.func
};

Progress.defaultProps = {
  progress: 0,
  type: 'dots',
  steps: [],
  onChange: () => {}
};

export default Progress;
