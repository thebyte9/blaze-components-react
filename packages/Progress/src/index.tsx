import React, { useState } from "react";
type ProgressProps = {
  progress: number,
  type?: string | any,
  steps: any[],
  onChange: (...args: any[]) => any
};
const Progress: React.SFC<ProgressProps> = ({
  progress,
  steps,
  type,
  onChange
}) => {
  const [_progress, setProgress] = useState(progress);
  const handleClick = ({ event, step }: { event: any, step: any }) => {
    setProgress(step);
    onChange({ event, step });
  };
  const checkStep = (step: any) => {
    if (step === steps.length && step === _progress) return "final current";
    if (step === _progress) return "current";
    if (step === steps.length) return "final";
    if (step <= _progress) return "visited";
    return "";
  };
  const isTypeCount = type === "count" ? "progress-bar__list-item--dots" : "";
  const modifiers = () => {
    const _modifiers = type.split(" ");
    const blockElement = "progress-bar__list-item--";
    if (!_modifiers.length) return `${blockElement}${type}`;
    return _modifiers.map((modifier: string) => `${blockElement}${modifier}`).join(" ");
  };
  return (
    <nav className="progress-bar">
      <ol className="progress-bar__list">
        {steps.map((text, index) => (
          <li
            key={text}
            className={`progress-bar__list-item ${modifiers()} ${isTypeCount} ${checkStep(
              index + 1
            )}`}
          >
            <span
              onClick={event => handleClick({ event, step: index + 1 })}
              role="button"
            >
              {text}
            </span>
          </li>
        ))}
      </ol>
    </nav>
  );
};
Progress.defaultProps = {
  progress: 0,
  type: "dots",
  steps: [],
  onChange: () => { }
};
export default Progress;