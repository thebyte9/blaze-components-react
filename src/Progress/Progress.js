import React, { useState } from 'react';
import PropTypes from 'prop-types';

const Progress = ({
  progress,
  steps,
  type,
  onChange
}) => {
  const [_progress, setProgress] = useState(progress);

  const handleClick = ({ e, i }) => {
    setProgress(i);
    onChange({ e, progress: i });
  };

  const checkStep = (i) => {
    if (i === steps.length && i === _progress) return 'final current';
    if (i === _progress) return 'current';
    if (i === steps.length) return 'final';
    if (i <= _progress) return 'visited';
    return '';
  };

  const isTypeCount = type === 'count' ? 'progress-bar__list-item--dots' : '';

  return (
    <nav className="progress-bar">
      <ol className="progress-bar__list">
        {steps.map((text, i) => (
          <li
          key={text}
          className={`progress-bar__list-item progress-bar__list-item--${type} ${isTypeCount} ${checkStep(i + 1)}`}>
            <span onClick={e => handleClick({ e, i: i + 1 })} role="button">{text}</span>
          </li>
        ))}
      </ol>
    </nav>
  );
};

Progress.propTypes = {
  progress: PropTypes.number,
  type: PropTypes.string,
  steps: PropTypes.array,
  onChange: PropTypes.func
};

Progress.defaultProps = {
  progress: 0,
  type: 'dots',
  steps: [],
  onChange: () => {}
};

export default Progress;
