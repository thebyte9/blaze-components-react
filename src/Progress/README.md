
## Description

The progress bar, despite providing a good user experience, also helps users avoid frustration and successfully complete all the steps.

## Usage

* Progress dot indicators

```js
const steps = [
  'Cart', 'Billing', 'Delivery', 'Review & pay'
];

<Progress 
  progress={2} 
  onChange={({event, step}) => {}}
  steps={steps}/>
```

* Progress dot indicators + steps counter

```js
<Progress
   progress={3} 
   type="count" 
   onChange={({event, step}) => {}} 
   steps={steps}/>
```

* Progress text

```js
<Progress 
    progress={1} 
    type="text" 
    onChange={({event, step}) => {}}
    steps={steps}/>
```

* Progress + text + completed icon

```js
<Progress 
    progress={4} 
    type="text icon" 
    onChange={({event, step}) => {}} 
    steps={steps}/>
```

## API

##### Progress can receive a number of `props` as follow:


| NAME   | TYPE | DEFAULT | 
| :---  | :---:  | :---: | 
| steps | Array | [] | 
| progress | Number | 0 | 
| type | Text | dots  | 
| onChange | Function | () => {} |


