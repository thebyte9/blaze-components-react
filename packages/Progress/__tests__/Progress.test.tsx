import '@testing-library/jest-dom';

import Progress from '../src/Progress';
import React from 'react';
import { render } from '@testing-library/react';

const steps = ['Cart', 'Billing', 'Delivery', 'Review & pay'];

describe('Alert component', () => {
  test('should be defined and renders correctly (snapshot)', () => {
    const { asFragment } = render(<Progress progress={2} onChange={() => {}} steps={steps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should be type count and change progress', () => {
    const handleChange = ({ step }: any) => expect(step).toEqual(4);

    const { asFragment } = render(<Progress progress={1} type="count" onChange={handleChange} steps={steps} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
