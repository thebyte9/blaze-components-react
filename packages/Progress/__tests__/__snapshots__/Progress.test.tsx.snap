// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component should be defined and renders correctly (snapshot) 1`] = `[Function]`;

exports[`Alert component should be type count and change progress 1`] = `[Function]`;
